wazuh/wazuh-dashboard-plugins
# Replace discover threat hunting

### Description
This PR, adds and replaces the `events` tab on the `Threat hunting` module

Closes #6236 

<img width="1438" alt="Screenshot 2023-12-29 at 01 01 56" src="https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/e61d15b8-fb24-4a1b-b37e-c4522b52dee2">

# Tests

## Search bar

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Apply filters using search bar input | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using the search bar `add filter` feature | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using the date picker | :black_circle: | :black_circle: | :black_circle: |
| Apply filters with no results (Check the `No results match your search criteria` message)| :black_circle: | :black_circle: | :black_circle: |
| Use the `saved queries` feature (Check saved queries CRUD) | :black_circle: | :black_circle: | :black_circle: |

- Apply filters using search bar input

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/b8132277-61aa-4150-ab4e-6c4667a88fac

</details>

- Apply filters using the search bar `add filter` feature

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/56ceb8c9-c823-45c5-89ef-9b52562661ee

</details>

- Apply filters using the date picker

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/62f006ce-7df6-44ef-a9f0-bedaa080955d

</details>

- Apply filters with no results (Check the `No results match your search criteria` message)

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/001950f8-6492-4856-a547-c8d53250568f

</details>

- Use the `saved queries` feature (Check saved queries CRUD)

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/f6da1506-ec7b-4985-9ef3-7bfb132e3a55

</details>


## Histogram chart

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check histogram chart hits consistency | :black_circle: | :black_circle: | :black_circle: |
| Use chart options (three points: maximize, minimize, inspect) | :black_circle: | :black_circle: | :black_circle: |

- Check histogram chart hits consistency 

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/251f3b25-9580-4bf2-8f62-d752f2ffc58a

</details>


- Use chart options (three points: maximize, minimize, inspect) 

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/e70e8a5d-5ef7-4850-b1c8-1eb4cff9201b

</details>

## Data grid

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check all table header features (sort, show/hide columns, density, etc) | :black_circle: | :black_circle: | :black_circle: |
| Check table pagination and rows per page | :black_circle: | :black_circle: | :black_circle: |
| Check document detail flyout | :black_circle: | :black_circle: | :black_circle: |

- Check all table header features (sort, show/hide columns, density, etc)

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/97e0b055-dd6e-4f67-900b-c3131a5f5b5e

</details>


- Check table pagination and rows per page

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/43a39edf-c02d-4b77-80ac-e6e733ffc2ba

</details>

- Check document detail flyout

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/cc4ecca1-f184-4244-a960-e57664779181

</details>

### Check List
